### DISCUSSION TITLE
Using g3 as an alternative to bevy_transform

### DISCUSSION THREAD

**Discussion by wrnrlr** [bevyengine/bevy#5002] [Q&A]
Hi, I am the author of [g3](https://github.com/wrnrlr/g3), a library that uses geometric-algebra instead of linear-algebra for working in 3D projective space.

I want to make a plugin to plot scenes of points, lines, planes and meshes, similar to how this is done in the ganja in the  [coffeeshop](https://enkimute.github.io/ganja.js/examples/coffeeshop.html#pga3d_differentiation). I've been trying to do this in bevy and got a [basic example](https://github.com/wrnrlr/g3/blob/main/examples/differentiation.rs) working. 

However, the [implementation](https://github.com/wrnrlr/g3/blob/main/src/plot/plugin.rs) is a little silly because I always do the same calculation twice, first I transform an element (point, line or plane) with a transformation (translator, rotor, motor) and then detect these change and then update the bevy Transform to where the element should be based on it's location visavis the origin. Much rather I would like to update the vertexes of the mesh directly based on the value of my element.

Is there an example or way where I can keep using the bevy rendering pipeline but don't need to associate a scene and Transform for every element I want to display. The only other way option I can think of is just use bevy_ecs and build all the other stuff myself.

**Comment by JonahPlusPlus** [comment]
I believe you would have to fork it and rewrite it to use your math. Depending on the performance and ease-of-use, you may want to consider making a proposal for using PGA instead of the current implementation. You should also join the Discord, people reply faster there.
